src/LanguageServer/Protocol/Handler/CodeActions/CodeActionHelpers.cs

```diff
@@ -403,16 +410,48 @@ private static async ValueTask<ImmutableArray<UnifiedSuggestedActionSet>> GetAct
             return actionSets;
         }
 
-        private static CodeActionKind GetCodeActionKindFromSuggestedActionCategoryName(string categoryName)
+        private static CodeActionKind GetCodeActionKindFromSuggestedActionCategoryName(string categoryName, IUnifiedSuggestedAction suggestedAction)
             => categoryName switch
             {
                 UnifiedPredefinedSuggestedActionCategoryNames.CodeFix => CodeActionKind.QuickFix,
-                UnifiedPredefinedSuggestedActionCategoryNames.Refactoring => CodeActionKind.Refactor,
+                UnifiedPredefinedSuggestedActionCategoryNames.Refactoring => GetRefactoringKind(suggestedAction),
                 UnifiedPredefinedSuggestedActionCategoryNames.StyleFix => CodeActionKind.QuickFix,
                 UnifiedPredefinedSuggestedActionCategoryNames.ErrorFix => CodeActionKind.QuickFix,
                 _ => throw ExceptionUtilities.UnexpectedValue(categoryName)
             };
 
+        private static CodeActionKind GetRefactoringKind(IUnifiedSuggestedAction suggestedAction)
```

Comment:
Providing the proper type will have VS Code render the actions differently.
<img width="312" alt="image" src="https://github.com/user-attachments/assets/4c8bf9b8-8dce-40b0-b419-b567cd1cee55" />

<img width="347" alt="image" src="https://github.com/user-attachments/assets/a5d1c3fd-51ec-4980-ad0b-35295eff0f86" />